### DISCUSSION TITLE
Supported Version Life Cycle

### DISCUSSION THREAD

**Discussion by sebastianliebscher** [apache/superset#24813] [General]
Hi Superset Community, the [Version Life Cycle from Airflow](https://airflow.apache.org/docs/apache-airflow/stable/installation/supported-versions.html) inspired me to put this together, as I couldn't find any relevant info in this compressed format so far. If I missed any relevant info, please post them here. Maybe this can be further fed with information (as I assumed a lot here after reading the dev mailing list and GitHub comments) and even added to the [official documentation](https://superset.apache.org/docs/intro) or the [GitHub Wiki](https://github.com/apache/superset/wiki).

Version | Current Patch/Minor | State | First Release | Limited Support | EOL/Terminated
-- | -- | -- | -- | -- | --
2.1 | 2.1.0 | Supported | Apr 5, 2023 | TBD | TBD
2.0 | 2.0.1 | Supported | Jul 14, 2022| TBD | with the release of 4.0?
1.5 | 1.5.3 | Supported | Apr 28, 2022 | TBD | with the release of 3.0?
1.4 | 1.4.2 | EOL | Jan 19, 2022 | TBD | with the release of 2.0?

**Comment by michael-s-molina** [comment]
Hi @sebastianliebscher. We were just discussing this internally! We have a meeting scheduled for after 3.0 to review our release process and one topic is to establish a clear definition of supported releases, release cadence, documentation, etc. Thank you for this table, it will definitely help with discussions and it may help others while we wait for these definitions. I think it's fair to say we are currently only supporting 3.0 and 2.1 and there's no concept of Limited Support. A more accurate table would be:

Version | Current Patch/Minor | State | First Release | EOL/Terminated
-- | -- | -- | -- | --
3.0 | 3.0.0RC1 | Supported | TBD | TBD
2.1 | 2.1.0 | Supported | Apr 5, 2023 | TBD
2.0 | 2.0.1 | EOL | Jul 14, 2022 | Apr 5, 2023
1.5 | 1.5.3 | EOL | Apr 28, 2022 | Apr 5, 2023
1.4 | 1.4.2 | EOL | Jan 19, 2022 | Apr 28, 2022

[CVEs](https://superset.apache.org/docs/security/cves) for each of the releases above.

Pinging some folks who might be interested in this discussion: @rusackas @eschutho @john-bodley @dpgaspar

**Reply by sebastianliebscher** [reply]
Thank you @michael-s-molina for the quick feedback and clarification! This helps a lot!

**Reply by michael-s-molina** [reply]
I edited my comment to also include a link to the CVEs by release published by @dpgaspar.

**Comment by rusackas** [comment]
Indeed! As @michael-s-molina we're definitely looking forward to formalizing this, via the Release Strategy meetings (which you're welcome to join if you'd like), and probably Town Hall (which you can also join). These are both available on the [community calendar](https://superset.apache.org/community). Once we get it all ironed out, it should be on the docs and/or wiki, and probably on that calendar if we go with time-driven releases.

**Comment by ddhecker** [comment]
Have there been any updates on this? I don't see a formal end of support listed in the docs as was discussed in this thread. Thanks!

**Reply by rusackas** [reply]
I think the most current doc you'd want to refer to is the wiki:
https://github.com/apache/superset/wiki/Release-Process

Let us know if there are more specific answers/targets/dates you're looking for.
